### PR TITLE
fix compilation issues under Windows

### PR DESCRIPTION
**Pull request by elikosan**
With these fixes i can install torch7 with luarocks on Windows.
Compiler: Visual Studio 2015
OS: Windows 7 Pro

### CODE REVIEW

**Comment by fmassa** [comment]
Thanks for the fixes!
This seems quite similar to https://github.com/torch/torch7/pull/660 though.
@soumith maybe its time to merge one of the PRs ?

File: lib/TH/THMemoryFile.c [hunk 1]
```diff
@@ -479,7 +479,7 @@ static size_t THMemoryFile_writeLong(THFile *self, long *data, size_t n)
     size_t i;
     for(i = 0; i < n; i++)
     {
-      ssize_t nByteWritten;
+      size_t nByteWritten;
```
Comment (soumith) [inline lib/TH/THMemoryFile.c]: changing the ssize_t to size_t is likely breaking the contbuilds. See https://travis-ci.org/torch/torch7/builds/134239495

**Comment by soumith** [comment]
This looks better than #660 but fails contbuilds. made in-line comments.

**Comment by elikosan** [comment]
I just committed a fix to the ssize_t issue. Let's see if Travis likes it better...

**Comment by soumith** [comment]
Thanks! :)

**Pull request merged**